scripts/lib/aws_apigwv2_testutil.sh

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env bash
+
+###########################################
+# API
+###########################################
+create_http_api_and_validate() {
+## create api resource
+cat <<EOF | kubectl apply -f - >/dev/null 2>&1
+apiVersion: apigatewayv2.services.k8s.aws/v1alpha1
+kind: API
+metadata:
+  name: $api_name
+spec:
+  name: $api_name
+  protocolType: HTTP
+EOF
+
+sleep 10
+
+## validate that api-id was populated in resource status
+debug_msg "retrieve api-id from api/$api_name resource's status"
+api_id=$(kubectl get $api_resource_name -o=json | jq -r .status.apiID)
+
+if [[ -z "$api_id" ]];then
+	echo "FAIL: $api_resource_name resource's status does not have apiID"
+	exit 1
+fi
+
+## validate that api was created using apigatewayv2 get-api operation
+debug_msg "apigatewayv2 get-api with api-id $api_id"
+aws apigatewayv2 get-api --api-id="$api_id" > /dev/null 2>&1
+assert_equal "0" "$?" "Expected success from 'apigatewayv2 get-api --api-id=$api_id' but got $?" || exit 1
+}
+
+delete_http_api_and_validate() {
+#delete api resource
+debug_msg "delete api/$api_name resource"
+kubectl delete $api_resource_name >/dev/null 2>&1
+assert_equal "0" "$?" "Expected success from kubectl delete but got $?" || exit 1
+
+#validate that api was deleted using apigatewayv2 get-api operation
+debug_msg "get-api with api-id $api_id"
+aws apigatewayv2 get-api --api-id="$api_id" > /dev/null 2>&1
+assert_equal "254" "$?" "Expected not-found status code from 'apigatewayv2 get-api --api-id=$api_id' but got $?" || exit 1
```

Comment:
keep in mind that the aws CLI version 2 and aws CLI version 1 use different return codes unfortunately for a NotFound :(

Version 1 uses 255. version 2 uses 254, if I remember correctly.

So, you'll need to either:

1) create a new `assert_in` function that accepts multiple acceptable values or

2) Use an if statement like this:

https://github.com/aws/aws-controllers-k8s/blob/main/test/e2e/ecr/smoke.sh#L68-L72